src/Http.ts

```diff
@@ -55,6 +55,7 @@ export interface Response<Body> {
   }
   headers: Record<string, string>
   body: Body
+  xhr: XMLHttpRequest
```

Comment:
@bmazzarol @gcanti pointed out to me that this kind of change in `Response` type would be **breaking**, thus we could:
1. add it and bump the package's version to **0.6.x** (taking advantage of current [_pre-1.x_ version](https://semver.org/#spec-item-4))
2. do not change the `Response` interface, add the `response` object to the `XHR` type right side and use it `sendWithResp()`
```ts
interface ResponseAndXHR<A> {
  xhr: XMLHttpRequest
  response: Response<A>
}

type XHR<A> = Observable<Either<HttpError, ResponseAndXHR<A>>>

/**
 * Executes as `Cmd` the provided call to remote resource, mapping result to a `Msg` containing the full response and the original XMLHTTPRequest object.
 * @category utils
 * @since 0.5.8
 */
export function sendFull<A, Msg>(f: (e: Either<HttpError, ResponseAndXHR<A>>) => Msg): (req: Request<A>) => Cmd<Msg> {
  return req => xhr(req).pipe(map(result => T.of(O.some(f(result)))))
}

// --- Helpers
function toResult<A>(x: XHR<A>): Observable<Either<HttpError, A>> {
  return x.pipe(map(E.map(({ response }) => response.body)))
}

function xhr<A>(req: Request<A>): XHR<A> {
  return ajax(toXHRRequest(req)).pipe(
    map(aresp =>
      pipe(
        aresp,
        toResponse(req),
        decodeWith(req.expect),
        E.map(response => ({ response, xhr: aresp.xhr })) // <--- here
      )
    ),
    catchError((e: unknown): XHR<A> => of(E.left(toHttpError(req, e))))
  )
}
```
Second option is my favorite...

BTW I would manage this in a different PR

Reply:
Will create a new pr around the second option.